integration-tests/src/test/kotlin/org/stellar/anchor/platform/test/Sep31CustodyRpcEnd2EndTests.kt

```diff
@@ -0,0 +1,471 @@
+import org.stellar.reference.wallet.WalletServerClient
+import org.stellar.walletsdk.ApplicationConfiguration
+import org.stellar.walletsdk.StellarConfiguration
+import org.stellar.walletsdk.Wallet
+import org.stellar.walletsdk.anchor.MemoType
+import org.stellar.walletsdk.asset.IssuedAssetId
+import org.stellar.walletsdk.asset.StellarAssetId
+import org.stellar.walletsdk.horizon.SigningKeyPair
+import org.stellar.walletsdk.horizon.sign
+
+class Sep31CustodyRpcEnd2EndTests(
+  config: TestConfig,
+  val toml: Sep1Helper.TomlContent,
+  val jwt: String
+) {
+  private val gson = GsonUtils.getInstance()
+  private val walletSecretKey = System.getenv("WALLET_SECRET_KEY") ?: CLIENT_WALLET_SECRET
+  private val keypair = SigningKeyPair.fromSecret(walletSecretKey)
+  private val wallet =
+    Wallet(
+      StellarConfiguration.Testnet,
+      ApplicationConfiguration { defaultRequest { url { protocol = URLProtocol.HTTP } } }
+    )
+  private val maxTries = 90
+  private val anchorReferenceServerClient =
+    AnchorReferenceServerClient(Url(config.env["reference.server.url"]!!))
+  private val walletServerClient = WalletServerClient(Url(config.env["wallet.server.url"]!!))
+  private val sep12Client = Sep12Client(toml.getString("KYC_SERVER"), jwt)
+  private val sep31Client = Sep31Client(toml.getString("DIRECT_PAYMENT_SERVER"), jwt)
+  private val sep38Client = Sep38Client(toml.getString("ANCHOR_QUOTE_SERVER"), jwt)
+  private val platformApiClient =
+    PlatformApiClient(AuthHelper.forNone(), config.env["platform.server.url"]!!)
+
+  private fun compareAndAssertEvents(
+    asset: StellarAssetId,
+    expectedEvents: List<AnchorEvent>,
+    actualEvents: List<AnchorEvent>
+  ) {
+    expectedEvents.forEachIndexed { index, expectedEvent ->
+      actualEvents[index].let { actualEvent ->
+        expectedEvent.id = actualEvent.id
+        expectedEvent.transaction.id = actualEvent.transaction.id
+        expectedEvent.transaction.startedAt = actualEvent.transaction.startedAt
+        expectedEvent.transaction.updatedAt = actualEvent.transaction.updatedAt
+        expectedEvent.transaction.transferReceivedAt = actualEvent.transaction.transferReceivedAt
+        expectedEvent.transaction.completedAt = actualEvent.transaction.completedAt
+        expectedEvent.transaction.stellarTransactions = actualEvent.transaction.stellarTransactions
+        expectedEvent.transaction.memo = actualEvent.transaction.memo
+        expectedEvent.transaction.destinationAccount = actualEvent.transaction.destinationAccount
+        expectedEvent.transaction.customers = actualEvent.transaction.customers
+        expectedEvent.transaction.quoteId = actualEvent.transaction.quoteId
+        actualEvent.transaction.amountIn?.let {
+          expectedEvent.transaction.amountIn.amount = actualEvent.transaction.amountIn.amount
+          expectedEvent.transaction.amountIn.asset = asset.sep38
+          //          expectedEvent.transaction.amountIn.asset =
```

Comment:
Some fileds assignment is commented-out, is it expected?

Reply:
I removed this commented code